.changeset/tabaxi-fellas-inspire.md

```diff
@@ -0,0 +1,220 @@
+
+Note: these APIs require the use of [SchemaFactoryAlpha](https://fluidframework.com/docs/api/fluid-framework/schemafactoryalpha-class).
+
+> [!WARNING]
+> These APIs are in preview and are subject to change.
+> Until these APIs have stabilized, we do not recommend using them in production code.
+> We reserve the right to make breaking changes to these APIs, including their persisted data format.
+> Using these APIs in production code may result in data loss or corruption.
+
+#### Creating a table
+
+You can craft a table schema with `TableSchema.table`.
+Note that you will still be required to provide a schema for the cells that will appear in the table.
+
+```typescript
+class Table extends TableSchema.table({
+	schemaFactory,
+	cell: schemaFactory.string,
+}) {}
+
+const table = new Table({
+	columns: [{ id: "column-0" }],
+	rows: [{ id: "row-0", cells: { "column-0": "Hello world!" } }],
+});
+```
+
+#### Creating a table with custom column and row schema
+
+If you need to associate additional data with your rows or columns, you can customize your row and column schema via `TableSchema.column` and `TableSchema.row`.
+These schema can then be provided to `TableSchema.table`:
+
+```typescript
+const Cell = schemaFactory.string;
+
+class ColumnProps extends schemaFactory.object("TableColumnProps", {
+	/** The column label to display. */
+	label: schemaFactory.string,
+}) {}
+
+class Column extends TableSchema.column({
+	schemaFactory,
+	props: ColumnProps,
+}) {}
+
+class Row extends TableSchema.row({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+class Table extends TableSchema.table({
+	schemaFactory,
+	cell: Cell,
+	column: Column,
+	row: Row,
+}) {}
+
+const table = new Table({
+	columns: [
+		new Column({ props: { label: "Entry" } }),
+		new Column({ props: { label: "Date" } }),
+		new Column({ props: { label: "Amount" } }),
+	],
+	rows: [],
+});
+```
+
+#### Interacting with the table
+
+Table trees created using `TableSchema` offer various APIs to make working with tabular data easy.
+These include:
+
+- Insertion and removal of columns, rows, and cells.
+- Cell access by column/row.
+
+```typescript
+// Create an empty table
+const table = Table.empty();
+
+const column0 = new Column({
+	props: { label: "Column 0" },
+});
+
+// Append a column to the end of the table.
+table.insertColumn({
+	column: column0,
+});
+
+const rows = [
+	new Row({ cells: { } }),
+	new Row({ cells: { } }),
+];
+
+// Insert rows at the beginning of the table.
+table.insertRows({
+	index: 0,
+	rows,
+});
+
+// Set cell at row 0, column 0.
+table.setCell({
+	key: {
+		column: column0,
+		row: rows[0],
+	},
+	cell: "Hello",
+});
+
+// Set cell at row 1, column 0.
+table.setCell({
+	key: {
+		column: column0,
+		row: rows[1],
+	},
+	cell: "World",
+});
+
+// Remove the first row.
+// Note: this will also remove the row's cell.
+table.removeRow(rows[0]);
+
+// Remove the column.
+// Note: this will *not* remove the remaining cell under this column.
```

Comment:
I think it would be useful to have an example that allows for more than one column type, or at least one where the schema for the column and its use are not exactly the same semantically (for example: having a string column schema which you use for a description instead of using a description column for a description).

Using the same schema for two different columns would be a good way to show off this separation.

If its a bit much to include here, maybe just now the possibility and link the example app? I assume it does this kind of thing?

Reply:
I like the idea of linking to Nick's demo app. Though, because these APIs aren't published yet, it doesn't yet use them. Might be something to add when we promote these APIs farther?

An alternative might be to link to our internal "example" app, but I don't think we generally point users to those.

Reply:
I'll add a test case for this kind of scenario. Not sure if we'll want to add it here or not, but a unit test capturing the scenario seems valuable either way.

Reply:
You can link to the unit test. I think that would be fine for changeset, but longer lived docs can link Nicks example once its ready.

Reply:
I added a test and noted the scenario in the changeset